### PR TITLE
Added Allowed Apps Checkbox for Mobile Data

### PR DESCRIPTION
**Pull request by beimoe**
A feature to have selective app access depending on your connection (mobile / wifi). Can e.g. save mobile traffic for hungry applications like phone updates and so on.

### CODE REVIEW

**Comment by schwabe** [comment]
So what is the purpose of this PR? It decides if an app will use non-VPN or VPN connection depending on whether the VPN was started on wifi or mobile?

**Comment by beimoe** [comment]
> So what is the purpose of this PR? It decides if an app will use non-VPN or VPN connection depending on whether the VPN was started on wifi or mobile?

Yes, correct. Together with the new Android 11 Killswitch to forbid any data next to Vpn it has the purpose of similar "firewall" applications. There are other "pseudo" VPN firewall solutions for this, but 2 vpns at a time are not possible.

File: main/src/ui/res/layout/allowed_application_layout.xml [hunk 1]
```diff
@@ -22,35 +22,51 @@
...
+    <CheckBox
+        android:id="@+id/app_selected_mobile"
+        android:layout_width="48dp"
+        android:layout_height="48dp"
         android:layout_rowSpan="1"
+        android:layout_gravity="center_vertical"
+        android:layout_marginStart="8dip"
+        android:minWidth="48dp"
+        android:minHeight="48dp"
         android:visibility="visible" />
```
Comment (schwabe) [inline main/src/ui/res/layout/allowed_application_layout.xml:62]: This adds a second checkbox without any UI indication what it does.

**Comment by schwabe** [comment]
This function needs to have a very good explaination in the UI to understand what it does to be integrated in the app. And I think the behaviour that  an app is excluded from the if the VPN was connected on mobile but continues to be excluded when you roam to WiFi is weird and will be considered as a bug by most users.

File: main/src/ui/res/layout/allowed_application_layout.xml (hunk 1 above)
Comment (beimoe) [reply main/src/ui/res/layout/allowed_application_layout.xml:62]: Yes, what do you think about if the explanation would be above e.g. with an icon/text.

**Comment by beimoe** [comment]
That should not be the expected behavior. Since the client reconnects on wifi/mobile switching with the then correct allowed app list, the behavior on mobile or wifi is always the same and predictable.

**Comment by beimoe** [comment]
Oh, I accidently closed the request. If this is also what you wish, I have done it for you ;-)
If not, please reopen, thank you!

**Comment by schwabe** [comment]
I am just saying that this PR needs more polishing and fixing before I will accept, I am not refusing it. I am just not accepting it in the current form.

**Pull request closed without merging**